cluster_experiments/power_analysis.py

```diff
@@ -428,99 +580,73 @@ def from_config(cls, config: PowerConfig) -> "PowerAnalysis":
-                        f"Be specially careful when using switchback splitters, since the time splitter column is being overriden"
-                    )
-
-    def check_clusters(self):
-        has_analysis_clusters = hasattr(self.analysis, "cluster_cols")
-        has_splitter_clusters = hasattr(self.splitter, "cluster_cols")
-        not_cluster_cols_cond = not has_analysis_clusters or not has_splitter_clusters
-        assert (
-            not_cluster_cols_cond
-            or self.analysis.cluster_cols == self.splitter.cluster_cols
-        ), f"cluster_cols in analysis ({self.analysis.cluster_cols}) must be the same as cluster_cols in splitter ({self.splitter.cluster_cols})"
-
-        assert (
-            has_splitter_clusters
-            or not has_analysis_clusters
-            or not self.analysis.cluster_cols
-            or isinstance(self.splitter, RepeatedSampler)
-        ), "analysis has cluster_cols but splitter does not."
-
-        assert (
-            has_analysis_clusters
-            or not has_splitter_clusters
-            or not self.splitter.cluster_cols
-        ), "splitter has cluster_cols but analysis does not."
-
-        has_time_col = hasattr(self.splitter, "time_col")
-        assert not (
-            has_time_col
-            and has_splitter_clusters
-            and self.splitter.time_col not in self.splitter.cluster_cols
-        ), "in switchback splitters, time_col must be in cluster_cols"
-
     def check_inputs(self):
         self.check_covariates()
         self.check_treatment_col()
         self.check_target_col()
         self.check_treatment()
         self.check_clusters()
+        self.check_perturbator_consistency()
+
+    class AAPowerAnalysis(BasePowerAnalysis):
+        def __init__(
+            self,
+            splitter: RandomSplitter,
+            analysis: ExperimentAnalysis,
+            cupac_model: Optional[BaseEstimator] = None,
+            target_col: str = "target",
+            treatment_col: str = "treatment",
+            treatment: str = "B",
+            control: str = "A",
+            alpha: float = 0.05,
+            features_cupac_model: Optional[List[str]] = None,
+            seed: Optional[int] = None,
+            hypothesis: str = "two-sided",
+        ):
+            super().__init__(
+                splitter=splitter,
+                analysis=analysis,
+                cupac_model=cupac_model,
+                target_col=target_col,
+                treatment_col=treatment_col,
+                treatment=treatment,
+                control=control,
+                alpha=alpha,
+                features_cupac_model=features_cupac_model,
+                seed=seed,
+                hypothesis=hypothesis,
+            )
+
+            self.check_inputs()
+
+        def power_analysis(
+            self,
+            df: pd.DataFrame,
+            pre_experiment_df: Optional[pd.DataFrame] = None,
+            average_effect: Optional[float] = None,
```

Comment:
need to change to required

Reply:
why?

Reply:
hmmm I'm taking out the perturbator as the power analysis is based on a deterministic formula. The Optional I guess was due to the fact that you can specify the effect in the perturbator, as far as I saw in the comments through the code. Without perturbator, does it make sense to have it as optional? How can I get the power value if I pass no effect?

Reply:
ah, you're right